lib/src/block/block_encoder.rs

```diff
@@ -25,12 +24,12 @@ use super::symbol_statistics::{
 
 pub(crate) fn generate_block_data(
     input: &[u8],
+    rle_data: Vec<u8>,
```

Comment:
Did you intentionally use an owned `Vec`?

Reply:
I did that because `bwt(input: Vec<u8>)` also wants owned vector. Should I update that part as well?

Reply:
That might be a good measure!